assets/app/scripts/services/data.js

```diff
@@ -150,6 +150,9 @@ angular.module('openshiftConsole')
     this._getNamespace(resource, context, opts).then(function(ns){
       $http(angular.extend({
```

Comment:
if the passed opts contains `opts.http.auth`, it'll overwrite our requiresAuth flag (extend doesn't merge recursively)